src/adapters/index.ts

```diff
@@ -1,12 +1,33 @@
-import { Telegraf } from "telegraf";
+import { createClient } from "@supabase/supabase-js";
 import { BotConfig } from "../types";
-import { Adapters } from "../types/adapters";
-import { supabase } from "./supabase";
-export * from "./telegram";
+import { Access } from "./supabase/helpers/classes/access";
+import { Settlement } from "./supabase/helpers/classes/settlement";
+import { Label } from "./supabase/helpers/classes/label";
+import { Super } from "./supabase/helpers/classes/super";
+import { User } from "./supabase/helpers/classes/user";
+import { Wallet } from "./supabase/helpers/classes/wallet";
+import { Database } from "./supabase/types";
 
-export const createAdapters = (config: BotConfig): Adapters => {
+export function createAdapters(config: BotConfig) {
+  const client = generateSupabase(
+    config?.supabase?.url ?? process.env.SUPABASE_URL,
+    config?.supabase?.key ?? process.env.SUPABASE_KEY
+  );
   return {
-    supabase: supabase(config?.supabase?.url ?? process.env.SUPABASE_URL, config?.supabase?.key ?? process.env.SUPABASE_KEY),
-    telegram: new Telegraf(config?.telegram?.token ?? process.env.TELEGRAM_BOT_TOKEN).telegram,
+    supabase: {
+      client,
+
+      access: new Access(client),
+      wallet: new Wallet(client),
+      user: new User(client),
+      debit: new Settlement(client),
+      settlement: new Settlement(client),
+      label: new Label(client),
+      super: new Super(client),
```

Comment:
Pretty major refactoring where I tried to associate an object to handle every interaction with a table of its namesake.